hedera-node/hedera-app/src/main/java/com/hedera/node/app/config/internal/ConfigurationAdaptor.java

```diff
@@ -0,0 +1,359 @@
+        }
+        throw new NoSuchElementException(exceptionMessagePropertyDoesNotExist(name));
+    }
+
+    @Override
+    public String getValue(final String name, final String defaultValue) {
+        if (exists(name)) {
+            return getValue(name);
+        }
+        return defaultValue;
+    }
+
+    @Override
+    public <T> T getValue(final String name, final Class<T> type)
+            throws NoSuchElementException, IllegalArgumentException {
+        if (exists(name)) {
+            return propertySource.getTypedProperty(type, name);
+        }
+        throw new NoSuchElementException(exceptionMessagePropertyDoesNotExist(name));
+    }
+
+    @Override
+    public <T> T getValue(final String name, final Class<T> type, final T defaultValue)
+            throws IllegalArgumentException {
+        if (exists(name)) {
+            return getValue(name, type);
+        }
+        return defaultValue;
+    }
+
+    @Override
+    public List<String> getValues(final String name) {
+        if (exists(name)) {
+            return propertySource.getTypedProperty(List.class, name);
+        }
+        throw new NoSuchElementException(exceptionMessagePropertyDoesNotExist(name));
+    }
+
+    @Override
+    public List<String> getValues(final String name, final List<String> defaultValues) {
+        if (exists(name)) {
+            return getValues(name);
+        }
+        return defaultValues;
+    }
+
+    @Override
+    public <T> List<T> getValues(final String name, final Class<T> type)
+            throws NoSuchElementException, IllegalArgumentException {
+        if (exists(name)) {
+            return propertySource.getTypedProperty(List.class, name);
+        }
+        throw new NoSuchElementException(exceptionMessagePropertyDoesNotExist(name));
+    }
+
+    @Override
+    public <T> List<T> getValues(final String name, final Class<T> type, final List<T> defaultValues)
+            throws IllegalArgumentException {
+        if (exists(name)) {
+            return getValues(name, type);
+        }
+        return defaultValues;
+    }
+
+    /**
+     * This implementation only supports {@link GlobalConfig} and {@link NodeConfig} as config data records.
+     *
+     * @param type {@link GlobalConfig} or {@link NodeConfig}, otherwise a {@link IllegalArgumentException} will be
+     *             thrown
+     * @param <T>  type, must be {@link GlobalConfig} or {@link NodeConfig}
+     * @return the {@link GlobalConfig} or {@link NodeConfig} instance
+     * @see Configuration#getConfigData(Class)
+     */
+    @Override
+    public <T extends Record> T getConfigData(final Class<T> type) {
+        if (Objects.equals(type, GlobalConfig.class)) {
+            return (T) globalConfig;
+        }
+        if (Objects.equals(type, NodeConfig.class)) {
+            return (T) nodeConfig;
+        }
+        throw new IllegalArgumentException("Config data type '" + type.getName() + "' not defined");
```

Comment:
w.r.t. question I asked above here `IllegalArgumentException` is not in a `throws` clause ...